### ISSUE TITLE
Rename dd() to d() and Change Method Signature

### ISSUE THREAD

**Issue by prezire** [laravel/internals#934]
Would it be possible to rename dd() to just d(), and add a 2nd boolean parameter as exit or die, with a default value of true? Of course dd([...]) can be used at the bottom portion of a single file, however there are instances dd() will placed in multiple files. For example:
File 1:
 - d($a, false);
 - d($b, false);
File 2:
 - d($c, false);
 - d($d); //This stops the entire dump stack.

**Comment by mattstauffer** [comment]
I think you may be looking for ‘dump()’: https://laravel.com/docs/5.5/helpers#method-dump

**Comment by prezire** [comment]
Yeah, it's interesting we got dump() and dd() instead of adding a toggle parameter. But perhaps there's a valid reason for it. Thanks for the reply.

**Comment by drbyte** [comment]
`dd` = "dump and die"
`dump` = "dump"

They serve different purposes. 

I'd say it was intentional:
There's a general contemporary trend towards avoiding adding boolean parameters whose meaning is lost without studying the functions. Makes code more readable, without having to guess "what's that 2nd parameter for?" and "wait, should there be another parameter there, or not?".

**Comment by prezire** [comment]
I see. I hate that style of convenience as well where the methods perform something different based on what parameters you pass. That makes sense. Thanks for the clarification.

**Comment by sisve** [comment]
Do you still want this change, or can we close this issue?

**Comment by mfn** [comment]
Funny how many times I wished I had `d()` which would be like `dump()` because .. well, there's a reason it's not `dumpanddie()` => it's short! 😄

**Comment by prezire** [comment]
This is very subjective on my side in addition of being too lazy. For a moment there was a bit of enlightenment from drbyte. But hopefully the community or admins might want consistency between d(), dump() and dd(), and just use one. To let this move forward, I'd personally like to see d() implemented.

**Comment by mfn** [comment]
> I'd personally like to see d() implemented

I too like to see it, but I would have a different expectation: `d()` should just be dump, take any arguments to dump (and return them), so you can use it within code when passing arguments.

**Comment by drbyte** [comment]
I see consistency already between `dump()` and `dd()`.  I'm not sure any change is needed there. That's just my opinion, and how I use it. 

As for adding `d()`, I dunno. Not sure I'd use it: mostly cuz it's always only temporary. 

That said ...

It's easy to add a helper function to your own app. (Several Medium and StackOverflow and other blog articles about various approaches, over last several months.)

It's also easy to PR an addition to Laravel core. No guarantees it'll be accepted. But if you present your case and that case is deemed beneficial to core, then, well, chances are higher it'll be accepted.

If it's not something you're already using regularly because you've already added it to your own app, then maybe it's worth taking some time to implement it locally and see whether there are adaptations you discover you need.  ie: is making `d()` be an alias of `dump()` enough? or are you wanting to change its signature/return, and why ... etc.

**Comment by sisve** [comment]
We cannot rename these since it would be a huge cognitive change for everyone that has learned a debugging statement since years back.

We can add a new function, but a single-letter function in the global scope? That's like begging for conflicts.

**Comment by prezire** [comment]
Cool! So it's decided. Thanks for looking into this.

**Comment by afraca** [comment]
You can pass one or many arguments to both `dump` and `dd` , which makes the flag (to die or not) hard to work with. In Python you would have keyword arguments, in php not so much.
